To execute the default application inside the container, run:
singularity run --nv $CONTAINERDIR/pytorch-2.0.1.sif

This container is based on NGC 23.08
https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-23-08.html#rel-23-08
WARNING: Ignoring invalid distribution -riton-nightly (/sfs/qumulo/qhome/brc4cb/.conda/envs/falcon_40B/lib/python3.9/site-packages)
ERROR: Could not find a version that satisfies the requirement PIL (from versions: none)
ERROR: No matching distribution found for PIL
WARNING: Ignoring invalid distribution -riton-nightly (/sfs/qumulo/qhome/brc4cb/.conda/envs/falcon_40B/lib/python3.9/site-packages)
WARNING: Ignoring invalid distribution -riton-nightly (/sfs/qumulo/qhome/brc4cb/.conda/envs/falcon_40B/lib/python3.9/site-packages)
Downloading: "https://github.com/pytorch/vision/zipball/v0.10.0" to /home/brc4cb/.cache/torch/hub/v0.10.0.zip
/home/brc4cb/.conda/envs/falcon_40B/lib/python3.9/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/brc4cb/.conda/envs/falcon_40B/lib/python3.9/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=DenseNet201_Weights.IMAGENET1K_V1`. You can also use `weights=DenseNet201_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Downloading: "https://download.pytorch.org/models/densenet201-c1103571.pth" to /home/brc4cb/.cache/torch/hub/checkpoints/densenet201-c1103571.pth
  0%|          | 0.00/77.4M [00:00<?, ?B/s] 24%|██▎       | 18.3M/77.4M [00:00<00:00, 191MB/s] 50%|█████     | 39.0M/77.4M [00:00<00:00, 206MB/s] 77%|███████▋  | 59.8M/77.4M [00:00<00:00, 212MB/s]100%|██████████| 77.4M/77.4M [00:00<00:00, 210MB/s]
Traceback (most recent call last):
  File "/sfs/qumulo/qhome/brc4cb/ml3_assignment2/densenet_test.py", line 156, in <module>
    num_ftrs = model_ft.fc.in_features
  File "/home/brc4cb/.conda/envs/falcon_40B/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1695, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'DenseNet' object has no attribute 'fc'
